File: src/ManagementPlatform.Domain/PaymentTransaction.cs
namespace ManagementPlatform.Domain;

public sealed class PaymentTransaction
{
    public long Id { get; set; }
    public long CheckoutAttemptId { get; set; }
    public CheckoutAttempt CheckoutAttempt { get; set; } = null!;
    public PaymentStatus Status { get; set; }
    public int AttemptCount { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "USD";
    public string? ProviderTransactionId { get; set; }
    public string? FailureReason { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
}

File: src/main/java/com/managementplatform/domain/model/PaymentTransaction.java
package com.managementplatform.domain.model;

import com.managementplatform.domain.enums.PaymentStatus;
import java.math.BigDecimal;
import java.time.Instant;

public record PaymentTransaction(
    long checkoutAttemptId,
    PaymentStatus status,
    int attemptCount,
    BigDecimal amount,
    String currency,
    String providerTransactionId,
    String failureReason,
    Instant createdAt
) {
}
